server/src/grpc/mod.rs

```diff
@@ -46,30 +50,148 @@ pub enum GrpcFieldOption {
     Timestamp(Option<Timestamp>),
     None,
 }
-pub trait GrpcDataObjectType: prost::Message + Clone {
-    fn get_field_value(&self, key: &str) -> Result<GrpcField, ArrErr>;
-}
-pub trait GrpcObjectType<T>: prost::Message + Clone
+
+#[tonic::async_trait]
+pub trait GrpcObjectType<T, U>
 where
-    T: GrpcDataObjectType,
+    T: GenericObjectType<U> + PsqlResourceType + From<Id> + Clone + Sync + Send,
+    U: GrpcDataObjectType + TryFrom<Row>,
+    Status: From<<U as TryFrom<Row>>::Error>,
 {
-    fn get_id(&self) -> String;
-    fn get_uuid(&self) -> Result<Uuid, ArrErr> {
-        Uuid::parse_str(&self.get_id()).map_err(ArrErr::from)
+    async fn get_by_id<V>(&self, request: Request<Id>) -> Result<Response<V>, Status>
+    where
+        V: From<T>,
+    {
+        let id: Id = request.into_inner();
+        let mut resource: T = id.into();
+        let obj: Result<Row, ArrErr> = T::get_by_id(&resource.try_get_uuid()?).await;
+        if let Ok(obj) = obj {
+            resource.set_data(obj.try_into()?);
+            Ok(Response::new(resource.into()))
+        } else {
+            Err(Status::not_found("Not found"))
+        }
+    }
+
+    async fn get_all_with_filter<V>(
+        &self,
+        request: Request<SearchFilter>,
+    ) -> Result<Response<V>, Status>
+    where
+        V: TryFrom<Vec<Row>>,
+        Status: From<<V as TryFrom<Vec<Row>>>::Error>,
+    {
+        let filter: SearchFilter = request.into_inner();
+        let mut filter_hash = HashMap::<String, String>::new();
+        filter_hash.insert("column".to_string(), filter.search_field);
+        filter_hash.insert("value".to_string(), filter.search_value);
+
+        match T::search(&filter_hash).await {
+            Ok(rows) => Ok(Response::new(rows.try_into()?)),
+            Err(e) => Err(Status::new(Code::Internal, e.to_string())),
+        }
+    }
+
+    async fn insert<V>(&self, request: Request<U>) -> Result<Response<V>, Status>
+    where
+        T: From<U>,
+        U: 'async_trait,
+        V: From<GenericResourceResult<T, U>>,
+    {
+        let data = request.into_inner();
+        let mut resource: T = data.into();
+        grpc_debug!(
+            "Inserting new flight_plan with data {:?}",
```

Comment:
Isn't this supposed to be generic? Why is it flight_plan specific?

Reply:
Good catch, that's an old debug message, I should fix that.

Reply:
fixed